pkg/db/strategy.go

```diff
@@ -308,15 +309,27 @@ func (s *Strategy) getExisting(ctx context.Context, gvk schema.GroupVersionKind,
 }
 
 func (s *Strategy) Delete(ctx context.Context, obj types.Object) (types.Object, error) {
-	return s.Update(ctx, obj)
+	obj, err := s.Update(ctx, obj)
+	if err, ok := err.(*mysql.MySQLError); ok && err.Number == 1062 {
```

Comment:
nit: Can we leave a comment here or somewhere in this file that links to this PR? That should make any future investigation or edits for this issue easier to begin.

Reply:
Second this, if not a link to the PR then maybe a link to sql docs describing the error value meaning. I'm surprised there isn't some sort of human readable definition for these mysql error values